myhpom/forms/choose_network.py

```diff
@@ -15,22 +12,24 @@ def clean(self):
         custom_provider = cleaned_data.get('custom_provider')
         health_network = cleaned_data.get('health_network')
 
-        if ((not custom_provider) and (not health_network)) or (custom_provider and health_network):
-            raise forms.ValidationError(
-                'Please either choose a network or enter a custom network.'
-            )
+        if self.instance and self.instance.state and self.instance.state.healthnetwork_set.exists():
+            if (not custom_provider and not health_network) or (custom_provider and health_network):
+                raise forms.ValidationError(
+                    'Please either choose a network or enter a custom network.'
+                )
+        else:
+            if not custom_provider:
+                raise forms.ValidationError('Please enter a custom network.')
```

Comment:
This kinda looks funny on the screen since its only related to the `custom_provider`, but is presented as a generic validation error. I'd suggest associating this with `custom_provider`.

Reply:
I've edited the code to make it less funny (less nesting), but I think we should leave this as the generic validation error since that's how it's being used in the template.